src/contour/TerminalSessionManager.cpp

```diff
@@ -267,21 +335,36 @@ void TerminalSessionManager::removeSession(TerminalSession& thatSession)
         return;
     }
     _sessions.erase(i);
-    _app.onExit(thatSession); // TODO: the logic behind that impl could probably be moved here.
-
-    _previousActiveSession = [&]() -> TerminalSession* {
-        auto const currentIndex = getSessionIndexOf(_activeSession).value_or(0);
-        if (currentIndex + 1 < _sessions.size())
-            return _sessions[currentIndex + 1];
-        else if (currentIndex > 0)
-            return _sessions[currentIndex - 1];
-        else
-            return nullptr;
-    }();
-    managerLog()("Calculated next \"previous\" session index {}",
-                 getSessionIndexOf(_previousActiveSession).value_or(-1));
+    tryFindSessionForDisplayOrClose();
+    //_app.onExit(thatSession); // TODO: the logic behind that impl could probably be moved here.
```

Comment:
> the logic behind that impl could probably be moved here

Why?

`onExit` is a hook. Ok, the function it is in currently is called "removeSession", which does not indicate, that this session is actually to be closed/terminated. So yeah, we could move it. But then we should also do it, such that we can remove this line here completely :)

Reply:
this comment was here before :)